### ISSUE TITLE
App crash when returning from background if Dialog permission was showing in android side

### ISSUE THREAD

**Issue by DurankGts** [codenameone/CodenameOne#3496]
if  I request Dialog permission and next go to background and it return again the dialog is showed perfectly by if you repeat the same step again my app crash ang ANR is showed in android 10.

2021-09-16 16:52:16.837 1151-1151/? E/Layer: [Surface(name=AppWindowToken{3fb193 token=Token{ad5a682 ActivityRecord{28dadcd u0 com.geotracksolutionsint.forcesos/.ForceSOSStub t95737}}})/@0xd5f1df5 - animation-leash#0] No local sync point found

check this log in real device.

the methods star() , stop() doesn't fire if the Permission dialg is showed in fron of my app.

![image](https://user-images.githubusercontent.com/16245755/133684410-42af687b-5271-4fd7-97bb-d4fd30b2a04c.png)

**Comment by DurankGts** [comment]
@shannah this issue was solved with this committ?09a9c13

this committ is available today in this moment?

**Comment by DurankGts** [comment]
@shai-almog  please help me to check if this committ solved the issue in android side and if this committ is applied in the built server.

**Comment by DurankGts** [comment]
Please check the video about the behavior of you stub in the android side. As I describe up. When I go to background with the native dialog permission in front of my app many times. The focus of my app goes to black and next crashed. I put some log in your generate code by built server and I check that you have a synchronized code that when my app fail it keep waiting. Could you tell me why ?(see the video)

![image](https://user-images.githubusercontent.com/16245755/133861815-c8f2bb03-bfa8-454d-8786-e7bb4b8ea098.png)



[issue background.zip](https://github.com/codenameone/CodenameOne/files/7188754/issue.background.zip)

**Comment by DurankGts** [comment]
I test in android 10 and 11. in 10 fail like the video and with the 11 don't fail at the moment

**Comment by DurankGts** [comment]
Why some times if I have the native permission dialog in from of my app an I go to background. The method star() if fire instead of stop()?

**Comment by DurankGts** [comment]
I fix the problem ignoring this event that you generate when may app goes to backgrond.

**Comment by DurankGts** [comment]
I don't Know why you are sending this event if the app goes to backgrond

**Comment by DurankGts** [comment]
@shannah that related with this committ https://github.com/DurankGts/CodenameOne/commit/16ae9b8f59f08f9700bb5177dd66309c5f03b429

?

**Comment by DurankGts** [comment]
![image](https://user-images.githubusercontent.com/16245755/135868589-4e1858d0-b79c-41f7-a933-fa15d22f90a4.png)


this the way that I found correctly. I don't know why the build generated code in this specific case it send and start event to my app when I go to background.

**Comment by DurankGts** [comment]
@plumberg  do you have the same issue?

**Comment by plumberg** [comment]
@DurankGts Yes, I noticed that in Android 11 everything works fine. But in Android 10 when you reopen it from a background the app just didn't load the form (at least it happened few times. Now it works). 
When I debugged the behavior in Android Studio, I noticed that when you put the app in the background, sometimes it calls only stop() method, sometimes `start()` and `stop()` methods simultaneously. 
I wasn't able to debug the case when the app just doesn't load the form yet.
